### PR TITLE
Xeno strain purchase and evolving changes

### PR DESCRIPTION
**Pull request by Shad0vvs**
## Why It's Good For The Game

> add: Added basic checks to strain purchasing so you must be at full hp/not stunned/resting/in a stance/etc to purchase.

The above prevents strain purchasing (as there was only an isdead check before) while low hp/crit/etc so that you cannot abuse, for example; godcrest bug and/or purchasing acid runner strain to get out of crit.

> tweak: Made it so you cannot utilize abilities while evolving as a xeno.

The above prevents ability while in the evolve state so you cannot abuse, for example; pounce as a runner while evolving and then getting another pounce as a lurker.
Probably also good so a xeno in front of you does not quickly evolve in the middle of a fight while you're fighting it and get a free heal on a newmob.

## Changelog

:cl:
add: Added basic checks to strain purchasing so you must be at full hp/not stunned/resting/in a stance/etc to purchase.
add: Made it so you cannot utilize abilities while evolving as a xeno.
del: Removed the list mutators verb from xenos.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/XenoMutatorSets.dm [hunk 1]
```diff
@@ -26,12 +26,15 @@
 
 /datum/mutator_set/proc/list_and_purchase_mutators()
 	var/list/mutators_for_purchase = available_mutators()
+	var/mob/living/carbon/Xenomorph/X = usr
 	if(mutators_for_purchase.len == 0)
-		to_chat(usr, "You can't take another strain.")
+		to_chat(usr, "There are no avaliable strains.")
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/XenoMutatorSets.dm:31]:
```suggestion
		to_chat(usr, "There are no available strains.")
```

File: code/modules/mob/living/carbon/xenomorph/XenoMutatorSets.dm [hunk 2]
```diff
@@ -228,20 +231,34 @@
...
 		return //For some reason we don't have mutators
 	src.mutators.list_and_purchase_mutators()
 
-/mob/living/carbon/Xenomorph/verb/list_mutators()
-	set name = "List Strains"
-	set desc = "List your current Strain, if any."
-	set category = "Alien"
+/mob/living/carbon/Xenomorph/proc/strain_checks()
+	if(!check_state())
+		return
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/XenoMutatorSets.dm:242]: should this not be return false?

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Review by stanalbatross — APPROVED** [review]

File: code/modules/mob/living/carbon/xenomorph/XenoMutatorSets.dm [hunk 3]
```diff
@@ -26,12 +26,15 @@
 
 /datum/mutator_set/proc/list_and_purchase_mutators()
 	var/list/mutators_for_purchase = available_mutators()
+	var/mob/living/carbon/Xenomorph/X = usr
```
Comment (Watermelon914) [inline code/modules/mob/living/carbon/xenomorph/XenoMutatorSets.dm:29]: please make your variable names more detailed

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

**Review by stanalbatross — APPROVED** [review]
var name confirmed to be explicit

**Review by Geevies — APPROVED** [review]

**Pull request merged**